### PR TITLE
Add numeric ops and `define-fun`

### PR DESCRIPTION
**Pull request by rachitnigam**

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -277,19 +279,23 @@ impl Context {
         )
     }
 
+    /// Declares a unit value of the given sort
```
Comment (fitzgen) [inline src/context.rs:282]: It isn't a unit value but an unconstrained value, right?

And it seems fine to me to have sugar/helpers for common operations, even if you can technically implement the operation in terms of others.

**Review by fitzgen — COMMENTED** [review]
Most of this looks good to me, but I think the declare vs declare-const stuff should be left out until we come to consensus on #15.

File: src/context.rs (hunk 1 above)
Comment (rachitnigam) [reply src/context.rs]: Updated comments!

**Comment by rachitnigam** [comment]
Would it be possible to create a new release of the crate once this is merged?

**Review by fitzgen — APPROVED** [review]
Thanks!

**Comment by fitzgen** [comment]
> Would it be possible to create a new release of the crate once this is merged?

Done!

**Pull request merged**